src/puppetlabs/dujour/version_check.clj

```diff
@@ -72,7 +74,7 @@
    update-server-url :- schema/Str]
   (let [product-name (:product-name request-values)
         {:keys [group-id artifact-id]} (get-coords product-name)
-        current-version (version group-id artifact-id)
+        current-version (version group-id artifact-id (str (or (:version request-values) "")))
```

Comment:
I'd do the defaulting at a high level instead of pushing it into a lower function that seems like its only to extract a version from the jar